library/src/test/java/com/chuckerteam/chucker/internal/support/OkHttpUtilsTest.kt

```diff
@@ -66,4 +76,50 @@ internal class OkHttpUtilsTest {
 
         assertThat(mockRequest.isGzipped).isFalse()
     }
+
+    @Test
+    fun uncompressSource_withGzippedContent() {
+        val content = "Hello there!"
+        val source = Buffer()
+        GzipSink(source).buffer().use { it.writeUtf8(content) }
+
+        val result = source.uncompress(headersOf("Content-Encoding", "gzip"))
+            .buffer()
+            .use(BufferedSource::readUtf8)
+
+        assertThat(result).isEqualTo(content)
+    }
+
+    @Test
+    fun uncompressSource_withPlainTextContent() {
+        val content = "Hello there!"
+        val source = Buffer().writeUtf8(content)
+
+        val result = source.uncompress(headersOf())
+            .buffer()
+            .use(BufferedSource::readUtf8)
+
+        assertThat(result).isEqualTo(content)
+    }
+
+    @ParameterizedTest(name = "\"{0}\" must be supported: {1}")
+    @MethodSource("supportedEncodingSource")
+    @DisplayName("Check if body encoding is supported")
+    fun headersHaveSupportedEncoding(headers: Headers, isSupported: Boolean) {
+        val result = headers.hasSupportedContentEncoding
+
+        assertThat(result).isEqualTo(isSupported)
+    }
+
+    companion object {
+        @JvmStatic
+        fun supportedEncodingSource(): Stream<Arguments> = Stream.of(
+            "" to true,
```

Comment:
we lost `null to true` here. I supposed that's due to a Kotlin API right?

Reply:
Yup, brought it back.